SwiftUIProject/SwiftUIProject/Products/ProductsView.swift

```diff
@@ -0,0 +1,215 @@
+// Copyright © RoadMap. All rights reserved.
+
+import SwiftUI
+
+/// Products view
+struct ProductsView: View {
+    // MARK: - Constants
+
+    private enum Constants {
+        static let emptyText = ""
+        static let magnifyingglassImageName = "magnifyingglass"
+        static let searchBarText = "Search..."
+        static let cartImageName = "cart"
+        static let saleText = "79$"
+        static let minusText = "-"
+        static let plusText = "+"
+        static let totalPriceViewText = "Total price: $"
+        static let sofaIndex = 0
+        static let armchairIndex = 1
+        static let bedIndex = 2
+        static let chairIndex = 3
+        static let amountProductsText = "Amount products in basket: "
+    }
+
+    // MARK: - Public Properties
+
+    var body: some View {
+        VStack {
+            totalPriceView
+            ScrollView {
+                VStack {
+                    HStack {
+                        searchBarView
+                        purchaseButtonView
+                    }
+                    makeOneProductView(
+                        product: viewModel.furnitures[Constants.sofaIndex],
+                        imageWidth: 60
+                    )
+                    makeOneProductView(
+                        product: viewModel.furnitures[Constants.armchairIndex],
+                        imageWidth: 40
+                    )
+                    makeOneProductView(
+                        product: viewModel.furnitures[Constants.bedIndex],
+                        imageWidth: 60
+                    )
+                    makeOneProductView(
+                        product: viewModel.furnitures[Constants.chairIndex],
+                        imageWidth: 30
+                    )
+                }
+            }
+        }
+    }
+
+    // MARK: - Private Properties
+
+    @StateObject private var viewModel = ProductsViewModel()
+
+    private var searchBarView: some View {
+        ZStack {
+            Rectangle()
+                .foregroundColor(.white)
+                .overlay {
+                    RoundedRectangle(cornerRadius: 30)
+                        .stroke(.black)
+                        .foregroundColor(.white)
+                }
+            HStack {
+                Image(systemName: Constants.magnifyingglassImageName)
+                    .resizable()
+                    .foregroundColor(.gray.opacity(0.3))
+                    .frame(
+                        width: 20,
+                        height: 20
+                    )
+                    .fontWeight(.semibold)
+                Text(Constants.searchBarText)
+                    .foregroundColor(viewModel.customPurpleColor)
+                Spacer()
+            }
+            .padding()
+        }
+        .frame(
+            width: 300,
+            height: 50,
+            alignment: .leading
+        )
+        .padding(.vertical)
+    }
+
+    private var totalPriceView: some View {
```

Comment:
```suggestion
    private var totalPriceView: some View {
```
мне кажется все-таки нужно указывать что именно это за вью, тут наверняка Text, а если несколько по типу text/toggle можно указать Stack